src/web.rs

```diff
@@ -132,6 +132,32 @@ pub async fn create_identity(identity_form: Form<IdentityForm>, state: &State<Cl
     Status::Ok
 }
 
+#[post("/change", data = "<identity_form>")]
+pub async fn change_identity(identity_form: Form<IdentityForm>, state: &State<Client>) -> Status {
+    println!("Change identity");
+    
+    let mut my_identity: Identity;
+    if !Path::new(IDENTITY_FILE_PATH).exists() {
+        return Status::NotAcceptable;
+    } else {
+        my_identity = read_identity_from_file();
+    }
+
+    let identity_form: IdentityForm = identity_form.into_inner();
+    let mut identity_changes: Identity = Identity::new_empty();
+    identity_changes.name = identity_form.name;
+    identity_changes.company = identity_form.company;
+    identity_changes.email = identity_form.email;
+    identity_changes.postal_address = identity_form.postal_address;
+    
+    my_identity.update_from(&identity_changes);        
+    write_identity_to_file(&my_identity);
```

Comment:
- [ ] check for changes and check for empty value

I don't see you checking for identity changes and empty values here (if I'm wrong - correct me). This should be done both in frontend and in backend because imagine that a hacker will just download PostMan and start sending invalid values to the method and load the DHT with it.

Reply:
- [ ]  it should not be possible to edit node id. Please remove it.